src/modules/_nomodule/PixelManipulation.js

```diff
@@ -42,59 +42,112 @@ module.exports = function PixelManipulation(image, options) {
+          quality: 100
+        });
+
+        r.on('data', function (chunk) {
+          totalLength += chunk.length;
+          chunks.push(chunk);
+        });
+
+        r.on('end', function () {
+          var data = Buffer.concat(chunks, totalLength).toString('base64');
+          var datauri = 'data:image/' + options.format + ';base64,' + data;
+          if (options.output)
+            options.output(options.image, datauri, options.format);
+          if (options.callback) options.callback();
+        });
+      }
+      if (res) {
+        pixels = res;
+        generateOutput();
+      } else if (!options.extraManipulation) generateOutput();
+    }
+
+    if (!options.changePixel) extraOperation();
+
     if (options.changePixel) {
 
       /* Allows for Flexibility
        if per pixel manipulation is not required */
 
-      for (var x = 0; x < pixels.shape[0]; x++) {
-        for (var y = 0; y < pixels.shape[1]; y++) {
-          let pixel = options.changePixel(
-            pixels.get(x, y, 0),
-            pixels.get(x, y, 1),
-            pixels.get(x, y, 2),
-            pixels.get(x, y, 3),
-            x,
-            y
-          );
-
-          pixels.set(x, y, 0, pixel[0]);
-          pixels.set(x, y, 1, pixel[1]);
-          pixels.set(x, y, 2, pixel[2]);
-          pixels.set(x, y, 3, pixel[3]);
-
-          if (!options.inBrowser && !process.env.TEST) pace.op();
+      const imports = {
+        env: {
+          // paceOp: require('pace').op,
+          consoleLog: console.log,
+          perform: function (x, y) {
+            let pixel = options.changePixel(
+              pixels.get(x, y, 0),
+              pixels.get(x, y, 1),
+              pixels.get(x, y, 2),
+              pixels.get(x, y, 3),
+              x,
+              y
+            );
+
+            pixels.set(x, y, 0, pixel[0]);
+            pixels.set(x, y, 1, pixel[1]);
+            pixels.set(x, y, 2, pixel[2]);
+            pixels.set(x, y, 3, pixel[3]);
+          }
         }
+      };
+
+      const inBrowser = (options.inBrowser) ? 1 : 0;
+      const test = (process.env.TEST) ? 1 : 0;
+
+      if (options.inBrowser) {
+        
+        fetch('./manipulation.wasm').then(response =>
+          response.arrayBuffer()
+        ).then(bytes =>
+          WebAssembly.instantiate(bytes, imports)
+        ).then(results => {
+          results.instance.exports.manipulatePixel(pixels.shape[0], pixels.shape[1], inBrowser, test);
+          extraOperation();
+        }).catch(err => {
+          
+          for (var x = 0; x < pixels.shape[0]; x++) {
+            for (var y = 0; y < pixels.shape[1]; y++) {
+              let pixel = options.changePixel(
+                pixels.get(x, y, 0),
+                pixels.get(x, y, 1),
+                pixels.get(x, y, 2),
+                pixels.get(x, y, 3),
+                x,
+                y
+              );
+
+              pixels.set(x, y, 0, pixel[0]);
```

Comment:
Maybe a `map` could be interesting as well.
`var result = pixel.map((coor) => {pixels.set(x,y,coor)};);`